src/styles/main.scss

```diff
@@ -1,3 +1,76 @@
+$cell-size: 100px;
+$count-days: 31;
+$days-order: (mon, tue, wed, thu, fri, sat, sun);
+$gap: 1px;
+
+@mixin center-content {
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+}
+
 body {
   margin: 0;
 }
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  align-items: center;
+  gap: $gap;
+
+  max-width: calc(#{$cell-size} * 7 + 1px * 7 + 10px * 2);
+  margin: auto;
+
+  @include center-content;
+  .calendar__day {
+    position: relative;
+    box-sizing: border-box;
+    width: $cell-size;
+    height: $cell-size;
+
+    border: 1px solid black;
+    background-color: #eee;
+
+    font-size: 32px;
+    font-family: Arial, Helvetica, sans-serif;
+    transition: 0.5s;
+
+    &:hover {
+      cursor: pointer;
+      background-color: pink;
+      transform: translateY(-20px);
+    }
+
+    &::before {
+      @include center-content;
+
+      content: '';
+    }
+  }
+
+  @for $day from 1 through $count-days {
+    .calendar__day:nth-child(#{$day})::before {
+      content: '#{$day}';
+
+      @include center-content;
+    }
+  }
+
+  @each $start-day in $days-order {
+    &--start-#{$start-day} {
+      .calendar__day {
+        margin-left: ($start-day - 1) + ($cell-size + 1px + $gap);
+      }
+    }
+  }
+
+  @for $month-length from 28 through 31 {
+    &--month-length-#{$month-length} {
+      .calendar__day:nth-child(n + #{$month-length + 1}) {
+        display: none;
+      }
+    }
+  }
+}
```

Comment:
good practice is to leave 1 line at the end of the file (please do this fix in each your file)